Red docker pack integration tests

See, for example, https://github.com/tarantool/cartridge-cli/runs/5065220968?check_suite_focus=true

Fails are stable and reproduces on all Tarantool versions:
```
FAILED test/integration/pack/test_pack.py::test_custom_base_image_build_dockerfile[tgz]
```
for EE and
```
ERROR test/integration/pack/test_pack_docker.py::test_pack - AssertionError: ...
ERROR test/integration/pack/test_pack_docker.py::test_tarantool_uid_and_gid
```
for open-source.

--- a/cli/project/dockerfiles.go
+++ b/cli/project/dockerfiles.go
@@ -51,6 +51,7 @@ func GetBuildImageDockerfileTemplate(ctx *context.Ctx) (*templates.FileTemplate,
 
 	dockerfileParts = append(dockerfileParts,
 		baseLayers,
+		fixCentosEolRepo,
 		installBuildPackagesLayers,
 		installTarantoolLayers,
 		wrapUserLayers,
@@ -83,7 +84,7 @@ func GetRuntimeImageDockerfileTemplate(ctx *context.Ctx) (*templates.FileTemplat
 			return nil, fmt.Errorf("Failed to get install Tarantool Dockerfile layers: %s", err)
 		}
 
-		dockerfileParts = append(dockerfileParts, createTarantoolUser, installTarantoolLayers)
+		dockerfileParts = append(dockerfileParts, createTarantoolUser, fixCentosEolRepo, installTarantoolLayers)
 	} else {
 		dockerfileParts = append(dockerfileParts, createTarantoolUser, createTarantoolDirectories)
 	}
@@ -248,6 +249,14 @@ USER {{ .TarantoolUID }}:{{ .TarantoolGID }}
 ENV CARTRIDGE_RUN_DIR=/var/run/tarantool
 ENV CARTRIDGE_DATA_DIR=/var/lib/tarantool
 ENV TARANTOOL_INSTANCE_NAME=default
+`
+
+	fixCentosEolRepo = `### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
 `
 
 	installTarantoolOpensourceLayers = `### Install opensource Tarantool

--- a/cli/project/dockerfiles_test.go
+++ b/cli/project/dockerfiles_test.go
@@ -165,6 +165,13 @@ func TestGetBuildImageDockerfileTemplateEnterprise(t *testing.T) {
 
 	expLayers = `FROM centos:7
 
+### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
+
 ### Install packages required for build
 RUN yum install -y git-core gcc gcc-c++ make cmake unzip
 
@@ -201,6 +208,13 @@ RUN yum install -y zip
 
 	expLayers = `FROM centos:7
 RUN yum install -y zip
+
+### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
 
 ### Install packages required for build
 RUN yum install -y git-core gcc gcc-c++ make cmake unzip
@@ -249,6 +263,13 @@ func TestGetBuildImageDockerfileTemplateOpensource(t *testing.T) {
 
 	expLayers = `FROM centos:7
 
+### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
+
 ### Install packages required for build
 RUN yum install -y git-core gcc gcc-c++ make cmake unzip
 
@@ -285,6 +306,13 @@ RUN yum install -y zip
 
 	expLayers = `FROM centos:7
 RUN yum install -y zip
+
+### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
 
 ### Install packages required for build
 RUN yum install -y git-core gcc gcc-c++ make cmake unzip
@@ -452,6 +480,13 @@ func TestGetRuntimeImageDockerfileTemplateOpensource(t *testing.T) {
 RUN groupadd -r -g {{ .TarantoolGID }} tarantool \
     && useradd -M -N -l -u {{ .TarantoolUID }} -g tarantool -r -d /var/lib/tarantool -s /sbin/nologin \
         -c "Tarantool Server" tarantool
+
+### Fix CentOS 8 EOL repo
+RUN if grep -q "CentOS Linux 8" /etc/os-release; then \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/mirrorlist=/#mirrorlist=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's/#baseurl=/baseurl=/g' {} + ; \
+        find /etc/yum.repos.d/ -type f -exec sed -i 's|mirror.centos.org|linuxsoft.cern.ch/centos-vault|g' {} + ; \
+    fi
 
 ### Install opensource Tarantool
 RUN curl -L https://tarantool.io/installer.sh | VER=1.10 bash -s -- --type release \

